lib/src/arithmetic/values/fixed_point_value.dart

```diff
@@ -0,0 +1,167 @@
+      {required bool signed, required int m, required int n}) {
+    if (value.abs().floor() > pow(2, m) - 1) {
+      throw RohdHclException('value exceed integer part');
+    }
+    if ((!signed) & (value < 0)) {
+      throw RohdHclException('Negative value must be signed.');
+    }
+    final sign = value >= 0 ? LogicValue.zero : LogicValue.one;
+    final integerPart = value.abs().floor();
+    final fractionalPart = ((value.abs() - integerPart) * pow(2, n)).round();
+
+    return FixedPointValue(
+        sign: signed ? sign : LogicValue.empty,
+        integer: LogicValue.ofInt(integerPart, m),
+        fraction: LogicValue.ofInt(fractionalPart, n));
+  }
+
+  /// Returns the value of the fixed-point number in a Dart [double] type.
+  double toDouble() {
+    final value = integer.toInt().toDouble() +
+        (fraction.toInt().toDouble() / pow(2, fraction.width));
+    return (sign == LogicValue.empty) | (sign == LogicValue.zero)
+        ? value
+        : -value;
+  }
+
+  /// Returns a negative integer if `this` less than [other],
+  /// a positive integer if `this` greater than [other],
+  /// and zero if `this` and [other] are equal.
+  @override
+  int compareTo(Object other) {
+    if (other is! FixedPointValue) {
+      throw RohdHclException('Input must be of type FixedPointValue');
+    }
+    final thisValue = toDouble();
+    final otherValue = other.toDouble();
+    if (thisValue == otherValue) {
+      return 0;
+    } else if (thisValue < otherValue) {
+      return -1;
+    } else {
+      return 1;
+    }
+  }
+
+  /// Equal-to operation that returns a LogicValue.
+  LogicValue eq(FixedPointValue other) =>
+      compareTo(other) == 0 ? LogicValue.one : LogicValue.zero;
+
+  /// Not equal-to operation that returns a LogicValue.
+  LogicValue neq(FixedPointValue other) =>
+      compareTo(other) != 0 ? LogicValue.one : LogicValue.zero;
+
+  /// Less-than operation that returns a LogicValue.
+  LogicValue operator <(FixedPointValue other) =>
+      compareTo(other) < 0 ? LogicValue.one : LogicValue.zero;
+
+  /// Less-than operation that returns a LogicValue.
+  LogicValue operator <=(FixedPointValue other) =>
+      compareTo(other) <= 0 ? LogicValue.one : LogicValue.zero;
+
+  /// Less-than operation that returns a LogicValue.
+  LogicValue operator >(FixedPointValue other) =>
+      compareTo(other) > 0 ? LogicValue.one : LogicValue.zero;
+
+  /// Less-than operation that returns a LogicValue.
+  LogicValue operator >=(FixedPointValue other) =>
+      compareTo(other) >= 0 ? LogicValue.one : LogicValue.zero;
+
+  /// Addition operation that returns a FixedPointValue.
+  /// The result is signed if one of the operands is signed.
+  /// The result integer has the max integer width of the operands plus one.
+  /// The result fraction has the max fractional width of the operands.
+  FixedPointValue operator +(FixedPointValue other) {
+    final res = toDouble() + other.toDouble();
```

Comment:
doesn't a standard `LogicValue + LogicValue` (with sufficient extension/padding) give the same operation but support any arbitrary size and precision?  (similar comment for other operators)